general/src/protocol/contexts.rs

```diff
@@ -347,11 +353,198 @@ pub struct TraceContext {
     #[metastructure(max_chars = "enumlike")]
     pub op: Annotated<OperationType>,
 
+    /// Whether the trace failed or succeeded. Currently only used to indicate status of individual
+    /// transactions.
+    pub status: Annotated<TraceStatus>,
+
     /// Additional arbitrary fields for forwards compatibility.
     #[metastructure(additional_properties, retain = "true")]
     pub other: Object<Value>,
 }
 
+/// Trace status
+///
+/// Values from https://github.com/open-telemetry/opentelemetry-specification/blob/8fb6c14e4709e75a9aaa64b0dbbdf02a6067682a/specification/api-tracing.md#status
+/// Mapping to HTTP from https://github.com/open-telemetry/opentelemetry-specification/blob/8fb6c14e4709e75a9aaa64b0dbbdf02a6067682a/specification/data-http.md#status
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum TraceStatus {
```

Comment:
Can we ensure this enum will never have more than 255 entires? That's an Enum8 in clickhouse. It would be really good if we could avoid going past that number.

Reply:
Yes I statically verify this now.